### PR TITLE
Unit tests for all forget password and login view models

### PR DESCRIPTION
**Pull request by HassanTeslim007**
This PR contains unit tests for Forget Password new, Forget password OTP, Forget password Email, Login and OTP view model

I'm unable to make a screen record, so attached is a screenshot of all the passing tests.
unit tests

### CODE REVIEW

**Comment by jeremiahseun** [comment]
Your PR did not pass the flutter test build - Can you check the error given and fix?
Thank you.

File: test/view_models/login_viewmodel_test.dart [hunk 1]
```diff
@@ -8,13 +8,31 @@ void main() {
...
-    group('initialise -', () {
+    group('initialise/navigation -', () {
+      test('when called, load progressindicator if there is no user', () async {
+        final userService = getAndRegisterUserServiceMock();
+        final model = LoginViewModel();
+        var load = model.isLoading;
+        expect(load, userService.hasUser);
+      });
+
       test('When called, check if user is registered on the system', () async {
```
Comment (pauleke65) [inline test/view_models/login_viewmodel_test.dart:19]: store the user, then try to retrieve the user instead

File: test/view_models/otp_viewmodel_test.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    group('initialize -', () {
+      test('when called navigate to login view', () {
+        final service = getAndRegisterNavigationServiceMock();
+        final model = OTPViewModel();
+        model.navigateLogin;
+        verify(service.navigateTo(Routes.loginView));
+      });
+
+      test('when called, verify otp value is true,hence loading indicator is false',
+          () {
```
Comment (pauleke65) [inline test/view_models/otp_viewmodel_test.dart:21]: this test does not match the name

**Review by pauleke65 — CHANGES_REQUESTED** [review]
Changes Requested

**Comment by pauleke65** [comment]
Also run flutter analyze and flutter test and fix all errors

**Comment by HassanTeslim007** [comment]
I checked the error, some were due to unused imports and I've fixed them. The others are from failing tests in the connectivity service and those are not from my end. All my tests still passes.

**Comment by YenumJay** [comment]
Go through the comments and resolve accordingly

> I checked the error, some were due to unused imports and I've fixed them. The others are from failing tests in the connectivity service and those are not from my end. All my tests still passes.

File: test/view_models/otp_viewmodel_test.dart (hunk 2 above)
Comment (HassanTeslim007) [reply test/view_models/otp_viewmodel_test.dart:21]: Name Fixed

**Review by pauleke65 — APPROVED** [review]
LGTM

**Review by jeremiahseun — APPROVED** [review]
LGTM

**Pull request merged**